frontend/src/test/scala/bloop/bsp/BspProtocolSpec.scala

```diff
@@ -138,136 +131,133 @@ class BspProtocolSpec {
             case Left(error) => Left(error)
             case Right(options) =>
-              val optionItems0 = options.items
-              val optionItems = {
-                if (optionItems0.exists(_.target.isEmpty)) {
-                  Left(JsonRpcResponse.internalError(
-                    s"A scalac option item doesn't map to a target! $optionItems0"))
-                } else Right(optionItems0.map(i => (i.target.get.uri, i)).sortBy(_._1))
+              val uriOptions = options.items.map(i => (i.target.uri.value, i)).sortBy(_._1)
+              val expectedUriOptions = Project
+                .eagerLoadFromDir(configDir, logger.underlying)
+                .map(p => (p.bspUri.value, p))
+                .sortBy(_._1)
+
+              Assert
+                .assertEquals("Size of options differ", uriOptions.size, expectedUriOptions.size)
+              uriOptions.zip(expectedUriOptions).foreach {
+                case ((obtainedUri, opts), (expectedUri, p)) =>
+                  Assert.assertEquals(obtainedUri, expectedUri)
+                  val obtainedOptions =
+                    stringifyOptions(opts.options, opts.classpath, opts.classDirectory)
+                  val classpath = p.classpath.iterator.map(i => bsp.Uri(i.toBspUri)).toList
+                  val classesDir = bsp.Uri(p.classesDir.toBspUri)
+                  val expectedOptions =
+                    stringifyOptions(p.scalacOptions.toList, classpath, classesDir)
+                  Assert.assertEquals(obtainedOptions, expectedOptions)
               }
 
-              optionItems.map { uriOptions =>
-                val expectedUriOptions = Project
-                  .eagerLoadFromDir(configDir, logger.underlying)
-                  .map(p => (p.bspUri, p))
-                  .sortBy(_._1)
-
-                Assert
-                  .assertEquals("Size of options differ", uriOptions.size, expectedUriOptions.size)
-                uriOptions.zip(expectedUriOptions).foreach {
-                  case ((obtainedUri, opts), (expectedUri, p)) =>
-                    Assert.assertEquals(obtainedUri, expectedUri)
-                    val obtainedOptions =
-                      stringifyOptions(opts.options, opts.classpath, opts.classDirectory)
-                    val classpath = p.classpath.iterator.map(_.toBspUri).toList
-                    val expectedOptions =
-                      stringifyOptions(p.scalacOptions.toList, classpath, p.classesDir.toBspUri)
-                    Assert.assertEquals(obtainedOptions, expectedOptions)
-                }
-              }
+              Right(uriOptions)
           }
       }
     }
 
     reportIfError(logger) {
-      BspClientTest.runTest(bsp, configDir, logger)(c => clientWork(c))
+      BspClientTest.runTest(bspCmd, configDir, logger)(c => clientWork(c))
       ()
     }
   }
 
-  def testCompile(bsp: Commands.ValidatedBsp): Unit = {
+  def testCompile(bspCmd: Commands.ValidatedBsp): Unit = {
```

Comment:
It's unclear to me why this test takes `bspCmd` as argument and after makes many assumptions about the command that has been passed to it. Why don't you just construct the command inside the test? As it is, this function looks re-usable, but I don't think it is.

Reply:
Because it has to be platform independent